app/src/main/java/com/x8bit/bitwarden/data/platform/datasource/network/interceptor/ResponseJsonKeyTransformerInterceptor.kt

```diff
@@ -0,0 +1,37 @@
+package com.x8bit.bitwarden.data.platform.datasource.network.interceptor
+
+import com.x8bit.bitwarden.data.platform.util.parseToJsonElementOrNull
+import com.x8bit.bitwarden.data.platform.util.transformPascalKeysToCamelCase
+import kotlinx.serialization.json.Json
+import okhttp3.Interceptor
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+
+/**
+ * An OkHttp interceptor that transforms the JSON response body by converting all JSON object keys
+ * to camel case.
+ *
+ * This interceptor is useful for handling inconsistencies in JSON responses where key casing might
+ * vary.
+ */
+class ResponseJsonKeyTransformerInterceptor : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        val responseBody = response.body
+            ?: return response
+
+        val transformedJsonResponseBody = Json
+            .parseToJsonElementOrNull(responseBody.string())
+            ?.transformPascalKeysToCamelCase()
+            ?.toString()
```

Comment:
Are we OK with the performance impact this is going to have on network requests.

I did a little testing on the sync response and this takes anywhere from 12 to 200 milliseconds. This is not insignificant when the network request take about 450 milliseconds.

Reply:
Takes much longer with large vaults. See the attached videos for a side-by-side comparison. The memory usage spikes pretty high too. I'm not confident we won't see OOMEs come from this. 😢